diff --git a/tablefield.module b/tablefield.module
index 8f88022..26b4df6 100644
--- a/tablefield.module
+++ b/tablefield.module
@@ -635,6 +635,10 @@ function tablefield_theme() {
  * Theme function for table view
  */
 function theme_tablefield_view($variables) {
+  // don't dispaly empty table
+  if (!$variables['rows']) {
+    return '';
+  }
   $attributes = array(
     'id' => 'tablefield-' . $variables['delta'],
     'class' => array(
